src/rgw/rgw_sal_motr.h

```diff
@@ -40,6 +43,177 @@ class MotrStore;
 #define RGW_MOTR_BUCKET_HD_IDX_NAME   "motr.rgw.bucket.headers"
 //#define RGW_MOTR_BUCKET_ACL_IDX_NAME  "motr.rgw.bucket.acls"
 
+// Implement watch-notify using Motr FDMI.
+// (1) A set of global indices, notify_watch_indices
+// (2) For each object to be watched, hash(obj_name) is used to pick which
+//     index in notifiy_watch_indices to insert a new key/value record,
+//     key = unique_fid(obj_name), value = marker (for fdmi filter) + notification msg.
```

Comment:
Sorry, I cannot get how this all mechanism works. How/where "object to be watched" is defined? Do we add a new key key/value to some motr index whenever the watched object is updated? Is this required by FDMI design or it's our implementation design? How/when this index is cleaned up?

Reply:
1. "object to be watched" means registering a callback function for an object. Actually, as the FDMI does not allow to register cb for a single object, the implementation doesn't define watch callback for each single object, the callback is for all objects. I will re-write the comment.
2. yes, we insert a key/value pair, but the key is the object's fid, that means each object is mapped to a key/value pair.
3. good point. As discussed with you and hua, I will change it by adding the fdmi marker to the value part of object meta index.
4. watch index is not cleaned up in this patch.